scripts/reorder_json_keys.py

```diff
@@ -0,0 +1,59 @@
+import json
+import os
+
+KEYS = [
+    "author",
+    "canonical_url",
+    "site_name",
+    "host",
+    "language",
+    "title",
+    "ingredients",
+    "ingredient_groups",
+    "instructions",
+    "instructions_list",
+    "category",
+    "yields",
+    "description",
+    "total_time",
+    "cook_time",
+    "prep_time",
+    "cuisine",
+    "cooking_method",
+    "ratings",
+    "ratings_count",
+    "equipment",
+    "reviews",
+    "nutrients",
+    "image",
+    "keywords",
+]
```

Comment:
Thanks @jknndy - this is nearly ready I think.

I'm trying to decide whether it's worth automating the collection of this field-list -- maybe by gathering the method names in the order that they appear in the template file.

It should be relatively rare to add/rename/remove the fields, but it'd be nice not to have to remember to update this script when we do.

Reply:
that would definitely be an improvement but I'm not currently seeing anywhere that it can be compiled from entirely. (this change could also be applied to `test_json_order` ).

We could update the `generate.py` file by pulling the `generate_scraper_test` function to use an external key dictionary with the full list on it and make the related changes. Then the list of keys could be pulled into other files as needed


Slightly related: thoughts if it would be worth implementing some testing for the generate file? some things to test would be the general functionality using `unittest.mock` & comparing the _possible new_ list of keys against the keys listed in `tests\__init__.py\ ... MANDATORY_TESTS & OPTIONAL_TESTS` ?

Reply:
> that would definitely be an improvement but I'm not currently seeing anywhere that it can be compiled from entirely. (this change could also be applied to `test_json_order` ).

Could the `AbstractScraper` class definition be suitable as the source?  (maybe we'd have to rearrange some of the methods in there, if so)